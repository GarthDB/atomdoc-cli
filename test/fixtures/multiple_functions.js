/**
 *  Private: If `node.type` is `Identifier` this function returns the type of the parent node.
 *
 *  * `node` {Object} Spidermonkey AST node.
 *
 *  Returns {String} the Spidermonkey node type.
 */
function _getNodeType(node) {
  return (node.type !== 'Identifier') ? node.type : node.parent.type;
}

/**
 *  Private: Checks `node.type` to see if it is a function node.
 *
 *  * `node` {Object} Spidermonkey AST node
 *
 *  Returns {Boolean} if it is a function node.
 */
function _nodeTest(node) {
  const nodeType = _getNodeType(node);
  const functionTypes = [
    'FunctionDeclaration',
    'ArrowFunctionExpression',
    'VariableDeclarator',
    'MethodDefinition',
  ];
  return functionTypes.includes(nodeType);
}

/**
 *  Public: Checks `node.type` to see if it is a function node.
 *
 *  * `node` {Object} Spidermonkey AST node
 *
 *  ## Examples
 *
 *  ```js
 *  const valid = nodeTest(node);
 *  ```
 *
 *  Returns {Boolean} if it is a function node.
 */
function nodeTest(node) {
  const nodeType = _getNodeType(node);
  const functionTypes = [
    'FunctionDeclaration',
    'ArrowFunctionExpression',
    'VariableDeclarator',
    'MethodDefinition',
  ];
  return functionTypes.includes(nodeType);
}
