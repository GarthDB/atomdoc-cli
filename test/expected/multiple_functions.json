[
  {
    "originalText": "Private: If `node.type` is `Identifier` this function returns the type of the parent node.\n\n* `node` {Object} Spidermonkey AST node.\n\nReturns {String} the Spidermonkey node type.",
    "visibility": "Private",
    "description": "If `node.type` is `Identifier` this function returns the type of the parent node.",
    "summary": "If `node.type` is `Identifier` this function returns the type of the parent node.",
    "arguments": [
      {
        "name": "node",
        "description": "{Object} Spidermonkey AST node.",
        "type": "Object",
        "isOptional": false
      }
    ],
    "returnValues": [
      {
        "type": "String",
        "description": "Returns {String} the Spidermonkey node type."
      }
    ],
    "start": 0,
    "end": 202,
    "name": "_getNodeType"
  },
  {
    "originalText": "Private: Checks `node.type` to see if it is a function node.\n\n* `node` {Object} Spidermonkey AST node\n\nReturns {Boolean} if it is a function node.",
    "visibility": "Private",
    "description": "Checks `node.type` to see if it is a function node.",
    "summary": "Checks `node.type` to see if it is a function node.",
    "arguments": [
      {
        "name": "node",
        "description": "{Object} Spidermonkey AST node",
        "type": "Object",
        "isOptional": false
      }
    ],
    "returnValues": [
      {
        "type": "Boolean",
        "description": "Returns {Boolean} if it is a function node."
      }
    ],
    "start": 306,
    "end": 476,
    "name": "_nodeTest"
  }
]
